src/qtt/algorithms/functions.py

```diff
@@ -337,20 +339,38 @@ def fit_gauss_ramsey(x_data, y_data, weight_power=0, maxiter=None, maxfun=5000,
 
     """
 
-    def func(params): return cost_gauss_ramsey(x_data, y_data, params, weight_power=weight_power)
+    def gauss_ramsey_model(x, amplitude, decay_time, frequency, phase, offset):
+        """  """
+        y = gauss_ramsey(x, [amplitude, decay_time, frequency, phase, offset])
+        return y
+
+    if weight_power is None:
+        weights = None
+    else:
+        diff_x = np.diff(x_data)
+        weights =  np.hstack( (diff_x[0], diff_x) )  ** weight_power
+        print(weights)
 
     if initial_params is None:
-        initial_params = estimate_parameters_damped_sine_wave(x_data, y_data, exponent=2)
+        initial_parameters = estimate_parameters_damped_sine_wave(x_data, y_data, exponent=2)
+    else:
+        initial_parameters = initial_params
+    lmfit_model = Model(gauss_ramsey_model)
+    lmfit_model.set_param_hint('amplitude', min=0)
+    lmfit_result = lmfit_model.fit(
+        y_data, x=x_data, **dict(zip(lmfit_model.param_names, initial_parameters)), verbose=verbose >= 2, weights = weights)
+
+    import qtt.algorithms.fitting
+    result_dict = qtt.algorithms.fitting.extract_lmfit_parameters(lmfit_model, lmfit_result)
 
-    fit_parameters = scipy.optimize.fmin(func, initial_params, maxiter=maxiter, maxfun=maxfun, disp=verbose >= 2)
+    result_dict['description'] = 'Function to analyse the results of a Ramsey experiment, fitted function: gauss_ramsey = ' \
+        'A * exp(-(x_data/t2s)**2) * sin(2pi*ramseyfreq * x_data - angle) +B'
 
-    result_dict = {
-        'description': 'Function to analyse the results of a Ramsey experiment, fitted function: gauss_ramsey = '
-                       'A * exp(-(x_data/t2s)**2) * sin(2pi*ramseyfreq * x_data - angle) +B',
-        'parameters fit': fit_parameters,
-        'parameters initial guess': initial_params}
+    # backwards compatibility
+    result_dict['parameters fit'] = result_dict['fitted_parameters']
```

Comment:
Are you sure parameters fit is without connecting  underscore?

Reply:
Somewhere else I found 'parameters fitted gaussian' as key value. I'm confused...

Reply:
The end goal is for all fitting methods to use the lmfit framework, and have all fitting methods use the sames names. For that there is a generic method: https://github.com/QuTech-Delft/qtt/blob/dev/src/qtt/algorithms/fitting.py#L30

With the line above I maintain backwards compatibility by also providing the field with the old name.

@sldesnoo-Delft @QFer

Reply:
I understand but I mean the backwards compatibiliy key here. Here you use 'parameters fit' but in https://github.com/QuTech-Delft/qtt/blob/2854e1ddedf4c10e471b5c69d4e246efd8d36ed7/src/qtt/algorithms/fitting.py#L247
the key 'parameters fitted gaussian' is used for storing the 'fitted_parameters'.
Are you sure this is the right backwards compatible key?

Reply:
This is a change of the `fit_gauss_ramsey`, the current keys are: https://github.com/QuTech-Delft/qtt/blob/dev/src/qtt/algorithms/functions.py#L350. So I think they are right. @QFer